ComPact/Verifier/PactVerifier.cs

```diff
@@ -53,13 +53,32 @@ public async Task VerifyPactAsync(string path)
 
             if (pact.Interactions != null)
             {
-                if (_config.ProviderBaseUrl == null)
-                {
-                    throw new PactException("Could not verify pacts. Please configure a ProviderBaseUrl.");
+                if (_config.ProviderBaseUrl == null && _config.ProviderHttpClient == null)
+                {
+                    throw new PactException("Could not verify pacts. Please configure a ProviderBaseUrl or a pre-configured ProviderHttpClient.");
                 }
-                using (var client = new HttpClient())
-                {
-                    tests.AddRange(await VerifyInteractions(pact.Interactions, _config.ProviderBaseUrl, client.SendAsync, _config.ProviderStateHandler));
+
+                var existingClient = _config.ProviderHttpClient != null;
+
+                try
+                {
+                    if (!existingClient)
+                    {
+                        _config.ProviderHttpClient = new HttpClient();
+                    }
+
+                    tests.AddRange(await VerifyInteractions(pact.Interactions, _config.ProviderBaseUrl, _config.ProviderHttpClient.SendAsync, _config.ProviderStateHandler));
+                }
+                finally
+                {
+                    if (!existingClient)
+                    {
+                        if (_config.ProviderHttpClient != null)
+                        {
+                            _config.ProviderHttpClient.Dispose();
+                            _config.ProviderHttpClient = null;
```

Comment:
I don't think this is necessary. A using block is a bit cleaner.

Reply:
Usually yes, but in this case it's a bit more complicated.
 - if we've created a HttpClient we should dispose it
 - if we've been given an HttpClient, we shouldn't dispose it because that's the concern of whatever created it, not ours.

I could re-write this block as:

```
if (_config.ProviderHttpClient == null)
{
    using (var client = new client)
    {
        tests.AddRange(await VerifyInteractions(pact.Interactions, _config.ProviderBaseUrl, client.SendAsync, _config.ProviderStateHandler));
    }
}
else
{
    tests.AddRange(await VerifyInteractions(pact.Interactions, _config.ProviderBaseUrl, _config.ProviderHttpClient.SendAsync, _config.ProviderStateHandler));
}
```

but in my opinion it's better to have a single call to tests.AddRange()? Happy to change it if you'd prefer it the other way though?

Reply:
> if we've been given an HttpClient, we shouldn't dispose it because that's the concern of whatever created it, not ours.

Good point.
If we want to simplify I think it would actually be fair to not dispose the client at all. After all it's just test code, so performance optimizations are very "nice-to-have"

Reply:
good point - i'll do that